wazuh/wazuh-splunk
# [RBAC] Implement services to get Splunk users

## Implement services to get Splunk users and current user logged in.
    
-  Implements a new backend controller that provides endpoints to retrieve:
    -  the user logged in the app
    - a list with all the users registered in the Splunk's instance.
- Implements a new frontend service (class) whom methods consume the new endpoints.

## Additions and deletions
### New files
- `SplunkAppForWazuh/appserver/controllers/users.py`
- `SplunkAppForWazuh/appserver/static/js/services/splunk-services/splunk-users.js`

### Modified files
- `SplunkAppForWazuh/default/web.conf`

### Formatted files
- `SplunkAppForWazuh/appserver/static/js/controllers/settings/api/settingsApiCtrl.js`
- `SplunkAppForWazuh/bin/get_agents_status.py`

## How to test
Load the new service 'splunkUsers'  on any view controller, for example on `settingsApiCtrl.js` and use it as follows:
```js
    constructor($scope, $currentDataService, apiList, $notificationService, $splunkUsers) {
      $splunkUsers.getCurrentUser()
      .then(data => console.log(data))

      $splunkUsers.getInternalUsers()
        .then(data => console.log(data))
    
      ....
    }
```
Finally navigate to the view on Splunk

The current user and the list of registered users should show on the browser's console.

![image](https://user-images.githubusercontent.com/15186973/137789748-e9b63b4c-b65b-4889-8d4d-c73543d0240e.png)
